packages/sitecore-jss-nextjs/src/edge/personalize-middleware.ts

```diff
@@ -81,10 +82,31 @@ export class PersonalizeMiddleware {
     }
   }
 
+  protected getExperienceContext(req: NextRequest): ExperienceContext {
```

Comment:
What do you think about moving that to the `personalize/cdp-service`? It's more related to the internal functionality of the cdp-service.
so `this.cdpService.executeExperience()` can accept
```js
const variantData = await this.cdpService.executeExperience(
      personalizeInfo.contentId,
      req,
      browserId
);
```
and there you will build `context` variable

And you will not log this variable twice using `debug`

Reply:
The `cdpService.executeExperience` is in `sitecore-jss` package so needs to remain agnostic as to the middleware implementation making the request (imagine us adding support for Personalize to other frameworks in the future). 

However, I totally agree there was too much duplicate logging going on. I've toned it down in the latest commit.